src/nodes/YAMLMap.ts

```diff
@@ -76,24 +76,30 @@ export class YAMLMap<K = unknown, V = unknown> extends Collection {
     }
   }
 
-  delete(key: K) {
+  delete(key: unknown): boolean {
     const it = findPair(this.items, key)
     if (!it) return false
     const del = this.items.splice(this.items.indexOf(it), 1)
     return del.length > 0
   }
 
-  get(key: K, keepScalar?: boolean) {
+  get(key: unknown, keepScalar: true): Scalar<V> | undefined
+  get(key: unknown, keepScalar?: boolean): V | undefined
```

Comment:
```suggestion
  get(key: unknown, keepScalar?: false): V | undefined
```
The general `boolean` case is caught in the next overload signature.